VimeoNetworking/Sources/VimeoClient.swift

```diff
@@ -176,125 +176,95 @@ final public class VimeoClient
+            
+            let failure: (URLSessionDataTask?, Error) -> Void = { (task, error) in
+                
+                DispatchQueue.global(qos: .userInitiated).async {
+
+                    self.handleTaskFailure(forRequest: request, task: task, error: error as NSError, completionQueue: completionQueue, completion: completion)
+                }
             }
-        }
-        
-        let path = request.path
-        let parameters = request.parameters
-        
-        let task: URLSessionDataTask?
-        
-        switch request.method
-        {
-        case .GET:
-            task = self.sessionManager?.get(path, parameters: parameters, progress: nil, success: success, failure: failure)
-        case .POST:
-            task = self.sessionManager?.post(path, parameters: parameters, progress: nil, success: success, failure: failure)
-        case .PUT:
-            task = self.sessionManager?.put(path, parameters: parameters, success: success, failure: failure)
-        case .PATCH:
-            task = self.sessionManager?.patch(path, parameters: parameters, success: success, failure: failure)
-        case .DELETE:
-            task = self.sessionManager?.delete(path, parameters: parameters, success: success, failure: failure)
-        }
-        
-        guard let requestTask = task else
-        {
-            let description = "Session manager did not return a task"
             
-            assertionFailure(description)
+            let path = request.path
+            let parameters = request.parameters
             
-            let error = NSError(domain: type(of: self).ErrorDomain, code: LocalErrorCode.requestMalformed.rawValue, userInfo: [NSLocalizedDescriptionKey: description])
+            let task: URLSessionDataTask?
             
-            networkRequestCompleted = true
+            switch request.method
+            {
+            case .GET:
+                task = self.sessionManager?.get(path, parameters: parameters, progress: nil, success: success, failure: failure)
+            case .POST:
+                task = self.sessionManager?.post(path, parameters: parameters, progress: nil, success: success, failure: failure)
+            case .PUT:
+                task = self.sessionManager?.put(path, parameters: parameters, success: success, failure: failure)
+            case .PATCH:
+                task = self.sessionManager?.patch(path, parameters: parameters, success: success, failure: failure)
+            case .DELETE:
+                task = self.sessionManager?.delete(path, parameters: parameters, success: success, failure: failure)
+            }
             
-            self.handleTaskFailure(forRequest: request, task: task, error: error, completionQueue: completionQueue, completion: completion)
+            guard let requestTask = task else
+            {
+                let description = "Session manager did not return a task"
```

Comment:
nit: we could pass the string directly in the `assertionFailure` for consistency.